### PR TITLE
Consolidate se23 in the rest of stories

### PR DESCRIPTION
**Pull request by sophschneider**
Some `.stories` files had left over se23 logic. This cleans them up!

### CODE REVIEW

File: polaris-react/src/components/AppProvider/AppProvider.stories.tsx [hunk 1]
```diff
@@ -37,7 +35,7 @@ export function Default(_, context) {
         },
       }}
       features={{
-        polarisSummerEditions2023: context.globals.polarisSummerEditions2023,
+        polarisSummerEditions2023: true,
       }}
```
Comment (sophschneider) [inline polaris-react/src/components/AppProvider/AppProvider.stories.tsx:39]: This is temporary, the whole `feature` prop should be deleted in https://github.com/Shopify/polaris/issues/10189

File: polaris-react/src/components/Frame/Frame.stories.tsx [hunk 2]
```diff
@@ -365,7 +357,7 @@ function InAnApplicationComponent({
             },
           },
         }}
-        features={{polarisSummerEditions2023}}
+        features={{polarisSummerEditions2023: true}}
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.stories.tsx:360]: This is temporary, the whole `feature` prop should be deleted in `AppProvider` cleanup

File: polaris-react/src/components/Frame/Frame.stories.tsx [hunk 3]
```diff
@@ -717,7 +701,7 @@ function WithAnOffsetComponent({
...
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.stories.tsx:704]: This is temporary, the whole `feature` prop should be deleted in `AppProvider` cleanup

File: polaris-react/src/components/Frame/Frame.stories.tsx [hunk 4]
```diff
@@ -1082,7 +1060,7 @@ function WithSidebarEnabled({
...
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.stories.tsx:1063]: This is temporary, the whole `feature` prop should be deleted in `AppProvider` cleanup

**Review by laurkim — APPROVED** [review]
💯

**Pull request merged**